data_processing/eia.py

```diff
@@ -0,0 +1,2999 @@
+        'CS': 'Combined-Cycle Steam and Combustion',
+        'CT': 'Combined-Cycle Combustion',
+        'ES': 'Energy Storage',
+        'FC': 'Fuel Cell',
+        'FW': 'Flywheel',
+        'GT': 'Gas Turbine',
+        'HA': 'Hydrokinetic Axial Flow Turbine',
+        'HB': 'Hydrokinetic Wave Buoy',
+        'HK': 'Hydrokinetic Other',
+        'HY': 'Hydraulic Turbine',
+        'IC': 'Internal Combustion Engine',
+        'PS': 'Pumped Storage',
+        'OT': 'Other',
+        'ST': 'Steam Turbine',
+        'PV': 'Photovoltaic',
+        'WT': 'Wind Turbine Onshore',
+        'WT': 'Wind Turbine Offshore'
+    }
+
+    return mover
+
+
+def generation(year: str):
+    '''
+    This function generates a nested dictionary of the Universities
+    present in the EIA spreadsheet and their generation amounts
+    by each fuel/mover type combination.
+
+    Parameters:
+    -----------
+    year: str
+        This is the string for the year being analyzed.
+
+    Returns:
+    --------
+    final: dict
+        Nested dictionary of the Universities present in the EIA
+        spreadsheet and their generation amounts in MWhr by each fuel/mover
+        type combination.
+
+    '''
+
+    if not isinstance(year, str):
+
+        raise TypeError(
+            "Argument 'year' must be of type 'str'."
+        )
+
+    allowed_years = [
+        '2008',
+        '2009',
+        '2010',
+        '2011',
+        '2012',
+        '2013',
+        '2014',
+        '2015',
+        '2016',
+        '2017',
+        '2018'
+    ]
+
+    if year not in allowed_years:
+
+        raise ValueError(
+            f"Not a valid year. Possibilities include: {*allowed_years,}."
+        )
+
+    headers = {
+        '2008': 7,
+        '2009': 7,
+        '2010': 7,
+        '2011': 5,
+        '2012': 5,
+        '2013': 5,
+        '2014': 5,
+        '2015': 5,
+        '2016': 5,
+        '2017': 5,
+        '2018': 5
+    }
+
+    netGen = {
+        '2008': 'NET GENERATION (megawatthours)',
+        '2009': 'NET GENERATION (megawatthours)',
+        '2010': 'NET GENERATION (megawatthours)',
+        '2011': 'Net Generation (Megawatthours)',
+        '2012': 'Net Generation (Megawatthours)',
+        '2013': 'Net Generation (Megawatthours)',
+        '2014': 'Net Generation (Megawatthours)',
+        '2015': 'Net Generation (Megawatthours)',
+        '2016': 'Net Generation (Megawatthours)',
+        '2017': 'Net Generation (Megawatthours)',
+        '2018': 'Net Generation (Megawatthours)'
+    }
+
+    generation = pd.read_csv((path + year + '.csv'),
+                             header=headers[year],
+                             usecols=['Operator Name',
+                                      'NERC Region',
+                                      'NAICS Code',
+                                      'Reported Prime Mover',
+                                      'Reported Fuel Type Code',
+                                      str(netGen[year])])
```

Comment:
```suggestion
                                      'Net Generation (Megawatthours)'])
```

Making it consistent with previous suggestion